### DISCUSSION TITLE
Does running Firefox inside firejail provide the same level of isolation in comparison to a Virtual Machine?

### DISCUSSION THREAD

**Discussion by firejailaddssecuirty** [netblue30/firejail#4636] [Q&A] [answered]
I am running Firefox & all the other installed apps inside firejail. I mean each time I install a new app I run `sudo firecfg` to verify if its added to firejail. Now someone told me that a virtual machine provides better isolation than firejail. Is this really true ? I know that by default the Downloads directory is whitelisted & I am okay with that. Even when running Lubuntu inside Virtualbox I add a shared folder anyway.

So Firefox inside Virtualbox Vs Firefox running inside a full blown distro inside Virtualbox which provides better isolation ?
Or are both the solutions providing more or less the same level of isolation/security ?

**Comment by rusty-snake** [comment, marked as answer]
A VM has much more isolation then a sandbox like firejail or containers like docker.

Running firefox inside firejail will use the same basesystem, kernel, network namespace, hardware, ... while a VM will have it's on basesystem, kernel, network  namespace and virtual hardware.

TL;DR: VMs have much more isolation and security at the cost of a much higher resource consumption and lower convenience.

**Reply by firejailaddssecuirty** [reply]
Thanks for your unbiased answer. Fact is that's bad news for me. I have only 4GB of Ram & I am really short in cash right now so can't buy new Ram. I installed Lubuntu inside Virtualbox & then tried browsing the web using the VM but within like half an hour the host (Mint) started using swap & my desktop hanged. The only way out of that was using REISUB.

**Reply by rusty-snake** [reply]
I did not tried this yet, but maybe it works for you: https://katacontainers.io/.
